examples/gno.land/p/demo/acl/acl.gno

```diff
@@ -96,7 +118,117 @@ func (d *Directory) AddUserToGroup(user std.Address, group string) {
 	d.userGroups.Set(user.String(), groups)
 }
 
-// TODO: helpers to remove permissions.
-// TODO: helpers to adds multiple permissions at once -> {verbs: []string{"read","write"}}.
-// TODO: helpers to delete users from gorups.
-// TODO: helpers to quickly reset states.
+func (d *Directory) AddUserToGroups(user std.Address, groups []string) {
+	existing, ok := d.userGroups.Get(user.String())
+	var existingGroups []string
+	if ok {
+		existingGroups = existing.([]string)
+	}
+	existingGroups = append(existingGroups, groups...)
+	d.userGroups.Set(user.String(), existingGroups)
+}
+
+func (d *Directory) RemoveUserFromGroup(user std.Address, group string) {
+	existing, ok := d.userGroups.Get(user.String())
+	if !ok {
+		return
+	}
+	groups := existing.([]string)
+	for i, g := range groups {
+		if g == group {
+			groups = append(groups[:i], groups[i+1:]...)
+			break
+		}
+	}
+	d.userGroups.Set(user.String(), groups)
+}
+
+func (d *Directory) RemoveUserFromGroups(user std.Address, groups []string) {
+	existing, ok := d.userGroups.Get(user.String())
+	if !ok {
+		return
+	}
+	existingGroups := existing.([]string)
+	for _, group := range groups {
+		for i, g := range existingGroups {
+			if g == group {
+				existingGroups = append(existingGroups[:i], existingGroups[i+1:]...)
+				break
+			}
+		}
+	}
+	d.userGroups.Set(user.String(), existingGroups)
+}
+
+func (d *Directory) removePermsFromBucket(bucket string, p []perm) {
+	existing, ok := d.permBuckets.Get(bucket)
+	if !ok {
+		return
+	}
+	ps := existing.(perms)
+
+	for _, perm := range p {
+		for i, existingPerm := range ps {
+			if perm.Equal(existingPerm) {
+				ps = append(ps[:i], ps[i+1:]...)
+				break
+			}
+		}
+	}
+
+	d.permBuckets.Set(bucket, ps)
```

Comment:
if you completely deleting the bucket when all permissions are removed, it seems possible to prevent storing empty entires.

```suggestion
	if len(ps) == 0 {
		d.permBuckets.Remove(bucket)
	} else {
		d.permBuckets.Set(bucket, ps)
	}
```